crates/nu-lsp/src/lib.rs

```diff
@@ -1147,7 +1177,7 @@ mod tests {
 
         open_unchecked(&client_connection, script.clone());
 
-        let resp = complete(&client_connection, script, 0, 14);
+        let resp = complete(&client_connection, script, 0, 13);
```

Comment:
I think this test should send that the client uses UTF-8 and the offsets should stay the same. Then there could be another test for UTF-16.

Reply:
I'm a little bit confused of our plan. Do you mean
1. enable utf-8 support in this PR, then keep both test cases
2. add another utf-16 test case and leave this utf-8 version ignored for later utf-8 support
3. others

Reply:
@schrieveslaach I pick plan 1 here in my latest commit, which leaves the client to choose which encoding to use. Take your time to review and test it.
I'd like to mention that there're some subtle corner cases here that I've no clear vision of how to solve:

1. neovim lsp doesn't support utf-8 very well. In my testing, goto def always treat returned ranges as utf-16, even if utf-8 is specified
2. mixed setups like

```lua
offset_encoding = "utf-8",
capabilities = {
  offset_encoding = { "utf-16" },
},
```
can still cause panic in server thread, better to detect it and provide useful error messages.